src/machines/java/com/enderio/machines/common/blockentity/PoweredSpawnerBlockEntity.java

```diff
@@ -0,0 +1,166 @@
+package com.enderio.machines.common.blockentity;
+
+import com.enderio.api.capability.StoredEntityData;
+import com.enderio.api.capacitor.CapacitorModifier;
+import com.enderio.api.capacitor.QuadraticScalable;
+import com.enderio.base.common.particle.RangeParticleData;
+import com.enderio.core.common.sync.BooleanDataSlot;
+import com.enderio.core.common.sync.EnumDataSlot;
+import com.enderio.core.common.sync.ResourceLocationDataSlot;
+import com.enderio.core.common.sync.SyncMode;
+import com.enderio.machines.common.blockentity.base.PoweredTaskMachineEntity;
+import com.enderio.machines.common.blockentity.task.SpawnTask;
+import com.enderio.machines.common.config.MachinesConfig;
+import com.enderio.machines.common.io.item.MachineInventoryLayout;
+import com.enderio.machines.common.lang.MachineLang;
+import com.enderio.machines.common.menu.PoweredSpawnerMenu;
+import net.minecraft.core.BlockPos;
+import net.minecraft.nbt.CompoundTag;
+import net.minecraft.network.chat.Component;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.server.level.ServerLevel;
+import net.minecraft.server.level.ServerPlayer;
+import net.minecraft.world.entity.player.Inventory;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.inventory.AbstractContainerMenu;
+import net.minecraft.world.level.block.entity.BlockEntityType;
+import net.minecraft.world.level.block.state.BlockState;
+import net.minecraft.world.phys.Vec3;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Optional;
+
+public class PoweredSpawnerBlockEntity extends PoweredTaskMachineEntity<SpawnTask> {
+
+    public static final QuadraticScalable CAPACITY = new QuadraticScalable(CapacitorModifier.ENERGY_CAPACITY, () -> 100000f);
+    public static final QuadraticScalable TRANSFER = new QuadraticScalable(CapacitorModifier.ENERGY_TRANSFER, () -> 200f);
+    public static final QuadraticScalable USAGE = new QuadraticScalable(CapacitorModifier.ENERGY_USE, () -> 160f);
+    private StoredEntityData entityData = StoredEntityData.empty();
+    private int range = 3;
+    private boolean rangeVisible;
+    protected float rCol;
+    protected float gCol;
+    protected float bCol;
+    private SpawnerBlockedReason reason = SpawnerBlockedReason.NONE;
+
+    public PoweredSpawnerBlockEntity(BlockEntityType type, BlockPos worldPosition, BlockState blockState) {
+        super(CAPACITY, TRANSFER, USAGE, type, worldPosition, blockState);
+        add2WayDataSlot(new BooleanDataSlot(this::isShowingRange, this::shouldShowRange, SyncMode.GUI));
+        addDataSlot(new ResourceLocationDataSlot(() -> this.getEntityType().orElse(new ResourceLocation("pig")),this::setEntityType, SyncMode.GUI));
```

Comment:
Just a stylistic point; maybe store the RL of the pig in a constant just so its abundantly clear (and we're also not creating new objects regularly)